### PR TITLE
Add Bitcoin Unlimited (1.0.1.1, current stable)

### PR DESCRIPTION
**Pull request by JustinAiken**
# Description

Pulled out of #31039, here's just the first new client

# Checklist

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [ ] `brew cask style --fix {{cask_file}}` reports no offenses.
  - rubocop-cask segfaults on me locally
- [x] The commit message includes the cask’s name and version.
- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not already refused in [closed issues].
- [x] Checked the cask is submitted to [the correct repo].

### CODE REVIEW

File: Casks/bitcoin-unlimited.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+cask 'bitcoin-unlimited' do
+  version '1.0.1.1'
+  sha256  '47aa38b237d15112ea41e70341be3d2cbbc0edc431a2ddf2533bb34cbd359db9'
+
+  url "https://www.bitcoinunlimited.info/downloads/bitcoinUnlimited-#{version}-osx.dmg"
+  name 'Bitcoin Unlimited'
+  homepage 'https://www.bitcoinunlimited.info'
+
+  conflicts_with cask: 'bitcoin-core',
```
Comment (vitorgalvao) [inline Casks/bitcoin-unlimited.rb]: You can already add the other ones you’ll be submitting. `conflicts_with` doesn’t yet work, anyway.

File: Casks/bitcoin-unlimited.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+cask 'bitcoin-unlimited' do
+  version '1.0.1.1'
+  sha256  '47aa38b237d15112ea41e70341be3d2cbbc0edc431a2ddf2533bb34cbd359db9'
+
+  url "https://www.bitcoinunlimited.info/downloads/bitcoinUnlimited-#{version}-osx.dmg"
```
Comment (vitorgalvao) [inline Casks/bitcoin-unlimited.rb]: These should not be aligned.

File: Casks/bitcoin-unlimited.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  conflicts_with cask: 'bitcoin-core',
+                 cask: 'bitcoin-xt'
+
+  depends_on     macos: '>= :mountain_lion'
+
+  # Renamed for consistency: app name is different in the Finder and in a shell.
+  app 'Bitcoin-Qt.app', target: 'Bitcoin Unlimited.app'
+
+  preflight do
+    set_permissions "#{staged_path}/Bitcoin-Qt.app", '0755'
```
Comment (vitorgalvao) [inline Casks/bitcoin-unlimited.rb]: Are you sure this is needed?

Comment (JustinAiken) [inline Casks/bitcoin-unlimited.rb]: It's not needed.  I'd just kept it in when I copied over the `bitcoin-core` cask, but it works fine without.  I can remove!

**Comment by JustinAiken** [comment]
Alright, green on CI after implementing your suggestions and rubocop's...  Do you want me to squash both your commits and mine into one?  Or just leave as is?

**Comment by vitorgalvao** [comment]
No need to squash. Github lets us do that on merge.

**Comment by JustinAiken** [comment]
Thanks! :)

**Pull request merged**